### ISSUE TITLE
Should UnneededSplatExpansion be Lint or Style?

### ISSUE THREAD

**Issue by jaredbeck** [rubocop-hq/rubocop#6796]
## Is your feature request related to a problem? Please describe.

It is my understanding that the "Lint" category identifies likely bugs, and the "Style" category does not. In other words, Lint cops are more "serious" than Style cops, right? I don't understand what type of bug `UnneededSplatExpansion` will prevent.

## Describe the solution you'd like

I think `UnneededSplatExpansion` should be in the "Style" category.

## Describe alternatives you've considered

Keep it in the "Lint" category.

## Additional context

Here's an example of code that produces an offense, but I don't understand why RuboCop thinks it is likely to cause a bug.

```ruby
user.attributes.slice(*%w[
  id
  name
  email
  birthdate
  # etc..
])
```

```plain
Offenses:

derp.rb:3:23: W: Lint/UnneededSplatExpansion: Pass array contents as separate arguments.
user.attributes.slice(*%w[ ...
                      ^^^^
```

**Comment by rrosenblum** [comment]
I feel like this discussion has come up before, however I am unable to find where. I think there is an argument to be made for both sides.

> It is my understanding that the "Lint" category identifies likely bugs, and the "Style" category does not. 

I share the same general understanding of the categories. I think we could expand this definition to include identifying ineffective or useless code. As far as I can tell, there is no formal definition of what lint is. Lint or linter was originally used to identify a piece of software that analyzes code for errors, bugs, and style issues. 

I think for this cop it partially depends on how you interpret what it is doing. Does it identify a style change or useless, less efficient, code?

I am fine with this cop being classified as either lint or style.

**Comment by bbatsov** [comment]
I'd rather keep it where it is just to spare the end users having to deal with one more breaking change. Especially if we don't have strong arguments to support the move.
